packages/x-adapter/src/endpoint-adapter.factory.ts

```diff
@@ -32,12 +32,12 @@ export const endpointAdapterFactory: EndpointAdapterFactory = <Request, Response
       requestMapper = identityMapper,
       responseMapper = identityMapper,
       defaultRequestOptions = {}
-    } = options;
+    }: EndpointAdapterOptions<Request, Response> = options;
 
     const endpoint = getEndpoint(rawEndpoint ?? requestEndpoint, request);
```

Comment:
The endpoints are in the wrong order. The request must override the options endpoint:

```suggestion
    const endpoint = getEndpoint(requestEndpoint ?? rawEndpoint, request);
```

Reply:
Okay, I've also refactored a test that I think was misleading regarding this behaviour